analysis_for_run.py

```diff
@@ -0,0 +1,236 @@
+                print('%s is "Complete" on BSSH' % args.run_name)
+                t=1
+                break
+        except:
+            continue
+        try:
+            if tmp.at[args.run_name,'Status']=='Failed' or tmp.at[args.run_name,'Status']=='Stopped' or tmp.at[args.run_name,'Status']=='Needs Attention' or tmp.at[args.run_name,'Status']=='Timed Out':
+                user=i
+                idd=tmp.at[args.run_name,'Id']
+                slack_message('%s has "Failed" or was unable to complete on BSSH; Script is aborting, check BSSH for more information' % args.run_name)
+                print('%s has "Failed" or was unable to complete on BSSH; Script is aborting, check BSSH for more information' % args.run_name)
+                sys.exit()
+        except:
+            continue
+    # Wait 20 min then look again.
+    if t==1:
+        break
+    time.sleep(1200)
+
+# Make dir that reads and anaylsis files will be saved to
+try:
+    os.makedirs("/Volumes/IDGenomics_NAS/WGS_Serotyping/%s/Sequencing_reads"% args.run_name)
+except:
+    print("Directory Already Exits, Make sure Reads are not already downloaded and delete Directory")
+    sys.exit()
+
+#If samples are directly avaiable in ICA there is no need to wait for download to start anaylsis by calling another python script
+if args.analysis == 'mycosnp':
+    subprocess.call("python auto_%s_ICA.py %s %s" % (args.analysis,args.run_name,args.ica_reference), shell=True)
+
+# Once the run is completed it will be downloaded
+# First the samplesheet must be downloaded
+bashCommand = """bs run download --config=%s
+          --output= "/Volumes/IDGenomics_NAS/WGS_Serotyping/%s/Sequencing_reads"
+          --id=%s
+          --extension=csv
+""" % (user,args.run_name,idd)
+
+process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+output, error = process.communicate()
+
+# Now the BSSH ids must be collected into a list
+bashCommand = "bs list dataset --config=%s --input-run=%s " % (user,idd)
+tmp=bs_out(bashCommand)
+for i in range(len(tmp)):
+    if tmp[i].split()[0]== 'Undetermined':
+        del tmp[i]
+        del tmp[i]
+        break
+tmp=pd.DataFrame(columns=tmp[0].split()[1:], data=[row.split()[1:4] for row in tmp[1:]])
+idds=list(tmp['Id'])
+
+# Download Reads from Run
+for i in idds:
+    bashCommand ="""
+    bs download dataset \
+            --config=%s \
+            --id=%s \
+            --output="/Volumes/IDGenomics_NAS/WGS_Serotyping/%s/Sequencing_reads/Raw" \
+            --retry
+    """ % (user,i,args.run_name)
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+print("Reads downloaded into: /Volumes/IDGenomics_NAS/WGS_Serotyping/%s/Sequencing_reads/Raw" % args.run_name)
+slack_message("Reads downloaded into: /Volumes/IDGenomics_NAS/WGS_Serotyping/%s/Sequencing_reads/Raw" % args.run_name)
+
+if args.download_reads:
+    sys.exit()
```

Comment:
This is so this script could be used just to download reads from basespace. I like this feature.